### DISCUSSION TITLE
MQTT mit TLS gesichert Zertifikat wird abgewiesen

### DISCUSSION THREAD

**Discussion by thb584Git** [tbnobody/OpenDTU#574] [Q&A] [answered]
A Mosquitto server is running on a Pi in my network already. Two self programmed ESP8266 units for monitoring and controlling a garage door are conected. Since the ability to control the door MQTT is running on Port 8883 and secured by TLS using a "self signed certificate".
This certificate is used also under Android by the App MQTT DASH and with Windows-SW MQTTfx to acces these topics
For the openDTU I get no connection up using this certificate, VSC-console shows this error message:
`18:27:06.404 > Connecting to MQTT...
18:27:06.499 > [1721252][E][ssl_client.cpp:37] _handle_error(): [start_ssl_client():278]: (-9984) X509 - Certificate verification failed, e.g. CRL, CA or signature check failed
18:27:06.518 > [1721256][E][WiFiClientSecure.cpp:135] connect(): start_ssl_client: -9984
18:27:06.521 > Disconnected from MQTT.
18:27:06.532 > Disconnect reason:TCP_DISCONNECTED`

OpenDTI Info/MQTT show this decrypted info
`cert. version : 3 serial number : 7......................:1D issuer name : C=DE, ST=Bavaria, L=Munich, O=Private, OU=Me, CN=pi-server subject name : C=DE, ST=Bavaria, L=Munich, O=Private, OU=Me, CN=pi-server issued on : 2021-04-14 14:02:42 expires on : 2031-04-12 14:02:42 signed using : RSA with SHA-256 RSA key size : 2048 bits basic constraints : CA=true`

With MQTTfx I need to configure the connection for using a "self signed certificate", how about openDTU? Anybody using MQTT secured be TLS? Any ideas?

**Comment by tbnobody** [comment]
Did you enter your CA certificate in OpenDTU? Thats a requirement to get the TLS connection working. (and by the way, the CA certificate is NOT the certificate of your PI)

**Comment by thb584Git** [comment]
Thx for the pompt reply. Well I replaced the default data on the OpenDTU/MQTT setup page after enabling TLS. The certificate info showed on the openDTU Info/MWTT page comes from my certificate as you can see so it seems to be accepted.
I created a set of certificates when I started the garage door project some some time ago:
`mosq_pi-server_ca.crt
mosq_pi-server_ca.der
mosq_pi-server_ca.key
mosq_pi-server_ca.srl
mosq_pi-server_ca_key.der
mosq_pi-server_serv.crt
mosq_pi-server_serv.csr
mosq_pi-server_serv.der
mosq_pi-server_serv.key`
I use the file named mosq_pi-server_ca.crt. 
I'm quite sure this is what I use on the Android devices. For MQTTfx I had to add 3 of the files to the configuation. The other .crt file  mosq_pi-server_serv.crt should belong to the server. However, running out of ideas,  I tried it either without success.

**Comment by tbnobody** [comment]
Did you also add the CA certificate to your mqtt broker configuration? The Broker needs the private key, the certificate and the CA certificate configured.

**Comment by tbnobody** [comment, marked as answer]
And I can imagine that it's also important that the hostname which you entered in the mqtt server hostname field in the OpenDTU configuration matches the CN (Common Name) in your server certificate.

**Comment by thb584Git** [comment]
You got it! I addessed it using the IP-address all the time, for not running into DNS-problems and missed this is a bit different here for MQTT. Changed ist to pi-server and is works.
Many thanks!! for your immediate help and time and in general for this great work to all contributors.
